spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/DefaultTaskExecutionService.java

```diff
@@ -210,6 +208,13 @@ public long executeTask(String taskName, Map<String, String> taskDeploymentPrope
 
 	@Override
 	public String getLog(String platformName, String taskId) {
+		Launcher launcher = this.launcherRepository.findByName(platformName);
+		// In case of Cloud Foundry, fetching logs by external execution Id isn't valid as the execution instance is destroyed.
+		// We need to use the task name instead.
+		if (launcher != null && launcher.getType().equals(TaskPlatformFactory.CLOUDFOUNDRY_PLATFORM_TYPE)) {
+			TaskDeployment taskDeployment = this.taskDeploymentRepository.findByTaskDeploymentId(taskId);
+			taskId = taskDeployment.getTaskDefinitionName();
```

Comment:
We get a NullPointerException if the user enters an invalid external-execution-id.   We should see an exception or something like what we get from the local deployer, "Log could not be retrieved as the task instance is not running."

Reply:
I'm afraid that this will have to be addressed in an additional PR as i've just merged this one

Reply:
I will have a PR to address this issue. 
@cppwfs Thank you for pointing this out.

Reply:
Here is the PR for this: https://github.com/spring-cloud/spring-cloud-dataflow/pull/3356. Closing this PR as it is already merged.